### PR TITLE
fix: Highlight keyword when search contacts

### PR DESCRIPTION
**Pull request by drminh2807**
https://github.com/linagora/twake-on-matrix/assets/12546908/201b93cd-27fa-4a03-8167-91221aa2cd4b

### CODE REVIEW

**Review by hoangdat — APPROVED** [review]

File: lib/pages/new_group/widget/contacts_selection_list.dart [hunk 1]
```diff
@@ -61,27 +62,48 @@ class ContactsSelectionList extends StatelessWidget {
...
+              final disabled = disabledContactIds.contains(
+                contact.matrixId,
+              );
+              return _ContactItem(
+                contact: contact,
+                selectedContactsMapNotifier: selectedContactsMapNotifier,
+                onSelectedContact: onSelectedContact,
+                highlightKeyword: success.keyword,
+                disabled: disabled,
+                paddingTop: index == 0 ? 8.0 : 0,
```
Comment (nqhhdev) [inline lib/pages/new_group/widget/contacts_selection_list.dart:76]: move it to `style`

File: lib/pages/new_group/widget/contacts_selection_list.dart [hunk 2]
```diff
@@ -38,8 +38,9 @@ class ContactsSelectionList extends StatelessWidget {
         ),
         (success) {
           if (success is PresentationExternalContactSuccess) {
-            return _buildContactItem(
-              context,
+            return _ContactItem(
+              selectedContactsMapNotifier: selectedContactsMapNotifier,
+              onSelectedContact: onSelectedContact,
               contact: success.contact,
               paddingTop: 8,
```
Comment (nqhhdev) [inline lib/pages/new_group/widget/contacts_selection_list.dart:45]: same

File: lib/widgets/highlight_text.dart [hunk 3]
```diff
@@ -6,13 +6,19 @@ class HighlightText extends StatelessWidget {
   final String? searchWord;
   final TextStyle? style;
   final TextStyle? highlightStyle;
+  final int? maxLines;
```
Comment (sherlockvn) [inline lib/widgets/highlight_text.dart:9]: should add default

Comment (imGok) [reply lib/widgets/highlight_text.dart:9]: these fields are nullable in Text.rich widget, it's fine imo

Comment (drminh2807) [reply lib/widgets/highlight_text.dart:9]: no need

File: lib/pages/new_group/widget/contacts_selection_list.dart (hunk 2 above)
Comment (drminh2807) [reply lib/pages/new_group/widget/contacts_selection_list.dart]: fixed

File: lib/pages/new_group/widget/contacts_selection_list.dart (hunk 1 above)
Comment (drminh2807) [reply lib/pages/new_group/widget/contacts_selection_list.dart]: fixed

**Review by sherlockvn — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**